File: src/main/templates/javascript_sdk/src/example/Product.js
/**
 * Product实体类 - 示例
 */
export class Product {
  constructor(data = {}) {
    this.id = data.id || null;
    this.name = data.name || null;
    this.description = data.description || null;
    this.price = data.price || null;
    this.stock = data.stock || null;
    this.category = data.category || null;
    this.status = data.status || null;
  }

  getId() {
    return this.id;
  }

  setId(id) {
    this.id = id;
  }

  getName() {
    return this.name;
  }

  setName(name) {
    this.name = name;
  }

  getDescription() {
    return this.description;
  }

  setDescription(description) {
    this.description = description;
  }

  getPrice() {
    return this.price;
  }

  setPrice(price) {
    this.price = price;
  }

  getStock() {
    return this.stock;
  }

  setStock(stock) {
    this.stock = stock;
  }

  getCategory() {
    return this.category;
  }

  setCategory(category) {
    this.category = category;
  }

  getStatus() {
    return this.status;
  }

  setStatus(status) {
    this.status = status;
  }

  /**
   * 转换为JSON对象
   */
  toJSON() {
    return {
      id: this.id,
      name: this.name,
      description: this.description,
      price: this.price,
      stock: this.stock,
      category: this.category,
      status: this.status
    };
  }

  /**
   * 从JSON对象创建Product实例
   */
  static fromJSON(data) {
    return new Product(data);
  }
}

File: src/main/templates/python_sdk/src/example/product.py
"""
Product实体类 - 示例
"""

from typing import Optional
from decimal import Decimal
from pydantic import BaseModel, Field


class Product(BaseModel):
    """Product实体类 - 示例"""
    
    id: Optional[str] = Field(None, description="产品ID")
    name: Optional[str] = Field(None, description="产品名称")
    description: Optional[str] = Field(None, description="产品描述")
    price: Optional[Decimal] = Field(None, description="价格")
    stock: Optional[int] = Field(None, description="库存数量")
    category: Optional[str] = Field(None, description="产品分类")
    status: Optional[str] = Field(None, description="产品状态")
    
    class Config:
        """Pydantic配置"""
        json_encoders = {
            Decimal: str,  # Decimal类型转换为字符串
        }
    
    def get_id(self) -> Optional[str]:
        """获取产品ID"""
        return self.id
    
    def set_id(self, product_id: Optional[str]) -> None:
        """设置产品ID"""
        self.id = product_id
    
    def get_name(self) -> Optional[str]:
        """获取产品名称"""
        return self.name
    
    def set_name(self, name: Optional[str]) -> None:
        """设置产品名称"""
        self.name = name
    
    def get_description(self) -> Optional[str]:
        """获取产品描述"""
        return self.description
    
    def set_description(self, description: Optional[str]) -> None:
        """设置产品描述"""
        self.description = description
    
    def get_price(self) -> Optional[Decimal]:
        """获取价格"""
        return self.price
    
    def set_price(self, price: Optional[Decimal]) -> None:
        """设置价格"""
        self.price = price
    
    def get_stock(self) -> Optional[int]:
        """获取库存数量"""
        return self.stock
    
    def set_stock(self, stock: Optional[int]) -> None:
        """设置库存数量"""
        self.stock = stock
    
    def get_category(self) -> Optional[str]:
        """获取产品分类"""
        return self.category
    
    def set_category(self, category: Optional[str]) -> None:
        """设置产品分类"""
        self.category = category
    
    def get_status(self) -> Optional[str]:
        """获取产品状态"""
        return self.status
    
    def set_status(self, status: Optional[str]) -> None:
        """设置产品状态"""
        self.status = status
    
    @classmethod
    def from_dict(cls, data: dict) -> 'Product':
        """从字典创建Product实例"""
        return cls(**data)
    
    def to_dict(self) -> dict:
        """转换为字典"""
        return self.dict()
